parser/charset/encoding.go

```diff
@@ -54,43 +52,34 @@ type Encoding struct {
 
 // enabled indicates whether the non-utf8 encoding is used.
 func (e *Encoding) enabled() bool {
-	return e.enc != nil && e.charLength != nil
+	return e != UTF8Encoding
 }
 
 // Name returns the name of the current encoding.
 func (e *Encoding) Name() string {
 	return e.name
 }
 
+// CharLength returns next chararacter length in bytes.
+func (e *Encoding) CharLength(bs []byte) int {
+	return e.charLength(bs)
+}
+
 // NewEncoding creates a new Encoding.
 func NewEncoding(label string) *Encoding {
 	if len(label) == 0 {
-		return &Encoding{}
+		return UTF8Encoding
 	}
-	e, name := Lookup(label)
-	if e != nil && name != encodingLegacy {
-		return &Encoding{
-			enc:         e,
-			name:        name,
-			charLength:  FindNextCharacterLength(name),
-			specialCase: LookupSpecialCase(name),
-		}
+
+	if e, exist := encodingMap[Format(label)]; exist {
+		return e
 	}
-	return &Encoding{name: name}
+	return UTF8Encoding
 }
 
 // UpdateEncoding updates to a new Encoding.
-func (e *Encoding) UpdateEncoding(label EncodingLabel) {
-	enc, name := lookup(label)
-	e.name = name
-	if enc != nil && name != encodingLegacy {
-		e.enc = enc
-		e.charLength = FindNextCharacterLength(name)
-	} else {
-		e.enc = nil
-		e.charLength = nil
-	}
-	e.specialCase = LookupSpecialCase(e.name)
+func UpdateEncoding(label string) *Encoding {
+	return NewEncoding(label)
```

Comment:
Please remove this function and use NewEncoding instead.

Reply:
updated